### PR TITLE
Merging thermistor library with master

### PR DESCRIPTION
**Pull request by vogeldylan**
Two new files added, analog_temp.c and analog_temp.h. Functions have been tested and confirmed working with the thermistor and ADC circuit. Includes testing code. 

Minor changes made to main.c, mostly cleaning up in advance of proper command handling development.

### CODE REVIEW

File: src/analog_temp.c [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+uint16_t read_thermistor_adc(int channel){
+    // Returns the raw ADC value for the specified channel
+    // Assumes channel is one of 0-6
+    uint16_t data;
+    uint8_t ADC_cmd, channel_bits;
+
+    if (channel % 2 == 0){
+        channel_bits = (uint8_t)(channel / 2);
+    } else {
+        channel_bits = 0b100  | (uint8_t)(channel / 2);
```
Comment (siddMahen) [inline src/analog_temp.c]: Not usually a good idea to do integer division like this. If `channel` is odd, `channel / 2` is always `1 = 0b001`. Thus `channel_bits` is always `0b101` in this case. Is this the intended behaviour?

File: src/analog_temp.c [hunk 2]
```diff
@@ -0,0 +1,113 @@
+/*
+
```
[the remark this answers is not in the document]
Comment (vogeldylan) [reply src/analog_temp.c]: Probably a good idea, although having the header at the top of the code is a good reminder to document, and accessible to anyone working on it. Tends to be an afterthought after I've finished a feature and scroll to the top of the file. 

Could be decided after the relevant documentation is created, and seeing whether this is made obsolete

File: src/analog_temp.c (hunk 1 above)
Comment (vogeldylan) [reply src/analog_temp.c]: No, the intended behavior is as follows (pg 10):
image

Comment (vogeldylan) [reply src/analog_temp.c]: Shouldn't `channel/2` return `0b000` for 1, `0b001` for 3, `0b010` for 5 ...?

File: src/analog_temp.c [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+    if (channel % 2 == 0){
+        channel_bits = (uint8_t)(channel / 2);
+    } else {
+        channel_bits = 0b100  | (uint8_t)(channel / 2);
+    }
+
+    ADC_cmd = THERMISTOR_BASE | (channel_bits << 4);
+
+    set_cs_low(COM_RST, &COM_RST_PORT);
+    data = send_spi(ADC_cmd);
```
[the remark this answers is not in the document]
Comment (vogeldylan) [reply src/analog_temp.c]: fixed

File: src/analog_temp.c (hunk 1 above)
Comment (siddMahen) [reply src/analog_temp.c]: Oh ofcourse; thanks, I'm being an idiot.

File: src/analog_temp.h [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+    DEPENDENCIES:       <uart/log.h>
+    AUTHORS:            Dylan Vogel
+    DATE MODIFIED:      2017-12-29
+
+###########################################################################
+
+    FUNCTION:           Prints the raw ADC value, calculated resistance,
+                        and calculated temperature to serial
+
```
[the remark this answers is not in the document]
Comment (vogeldylan) [reply src/analog_temp.h]: done

**Pull request merged**